packages/vite/src/node/utils.ts

```diff
@@ -323,7 +323,7 @@ export function injectQuery(url: string, queryToInject: string): string {
   // encode percents for consistent behavior with pathToFileURL
   // see #2614 for details
   let resolvedUrl = new URL(url.replace(/%/g, '%25'), 'relative:///')
-  if (resolvedUrl.protocol !== 'relative:') {
+  if (resolvedUrl.protocol === 'file:') {
     resolvedUrl = pathToFileURL(url)
   }
```

Comment:
Because we are not getting path part from the parsed URL, I guess we can remove this condition. refs #2435

Also this condition now doesn't make sense.
When `resolvedUrl.protocol` is `file:`, `url` has `file:` protocol. Then, we are passing file URL to `pathToFileURL`. This won't work.

/cc @poyoho

Reply:
I agree that this condition should be removed.

Reply:
@yzydeveloper Would you remove this condition?

Reply:
@sapphi-red  This condition has been removed. 5a4233092f864a65988f53e58a29fd594cbbc09c